python/evalica/__init__.py

```diff
@@ -998,6 +998,83 @@ def pairwise_frame(scores: pd.Series[float]) -> pd.DataFrame:
     return pd.DataFrame(pairwise_scores(scores.to_numpy()), index=scores.index, columns=scores.index)
 
 
+@dataclass
+class BootstrapConfidenceInterval:
+    score_method: Literal['elo', 'bradley-terry', 'newman']  # TODO: more?
+    num_rounds: int = 100
+    sample_rate: float = 1.0
+    with_replace: bool = True
+
+    def fit(self,
+            df: pd.DataFrame,
+            left_column: str = 'left',
+            right_column: str = 'right',
+            winner_colum: str = 'winner',
+            weights: Collection[float] | None = None,
+            win_weight: float = 1.0,
+            tie_weight: float = 0.5,
+            solver: Literal["naive", "pyo3"] = "pyo3",
+            **kwargs,
+            ) -> pd.DataFrame:
+        """
+        Calculate confidence interval by bootstrap.
+
+        Args:
+            df: The dataset with all the records.
+            left_column: The column name of left role.
+            right_column: The column name of right role.
+            winner_colum: The column name of winner role.
+            weights: The example weights.
+            win_weight: The win weight.
+            tie_weight: The tie weight.
+            solver: The solver.
+            kwargs: The method specific parameters(Like the `initial`, `base`, `scale`, `k` in elo method).
+
+        Returns:
+            The dataframe with scores from all bootstrap rounds.
+        """
+
+        score_function = self._get_score_method()
+        *_, index = indexing(
+            xs=df[left_column],
+            ys=df[right_column],
+        )
+
+        bootstrap: list["pd.Series[float]"] = []
+        for r in range(self.num_rounds):
+            df_sample = df.sample(frac=self.sample_rate, replace=self.with_replace, random_state=r)
+            result_sample = score_function(
+                xs=df_sample[left_column],
+                ys=df_sample[right_column],
+                winners=df_sample[winner_colum],
+                index=index,
+                weights=weights,
+                win_weight=win_weight,
+                tie_weight=tie_weight,
+                solver=solver,
+                **kwargs
+            )
+
+            bootstrap.append(result_sample.scores)
+
+        df_bootstrap = pd.DataFrame(bootstrap)
+        # TODO: calculate the quantiles in here?
```

Comment:
@dustalov I am not sure if we should calculate quantiles here, or use another method, any sugguestion?